lttoolbox/att_compiler.cc

```diff
@@ -90,90 +96,96 @@ AttCompiler::is_word_punct(wchar_t symbol)
  *         only) character otherwise.
  */
 int
-AttCompiler::symbol_code(const wstring& symbol)
+AttCompiler::symbol_code(const UString& symbol)
 {
-  if (symbol.length() > 1) {
+  if (u_strHasMoreChar32Than(symbol.c_str(), -1, 1)) {
     alphabet.includeSymbol(symbol);
     return alphabet(symbol);
-  } else if (symbol == L"") {
+  } else if (symbol.empty()) {
     return 0;
-  } else if ((iswpunct(symbol[0]) || iswspace(symbol[0])) && !is_word_punct(symbol[0])) {
-    return symbol[0];
   } else {
-    letters.insert(symbol[0]);
-    if(iswlower(symbol[0]))
-    {
-      letters.insert(towupper(symbol[0]));
+    UChar32 c = symbol[0];
+    if (symbol.size() > 1) {
+      vector<char> v8;
+      vector<UChar32> v32;
+      utf8::utf16to8(symbol.begin(), symbol.end(), std::back_inserter(v8));
+      utf8::utf8to32(v8.begin(), v8.end(), std::back_inserter(v32));
```

Comment:
This looks condensable to 1 conversion.

Reply:
it's also definitely working on exactly 1 codepoint, so likely excessive regardless